src/components/community/UpsertChatForm.tsx

```diff
@@ -49,6 +51,30 @@ export default function UpsertChatForm(props: UpsertChatFormProps) {
     props as UpsertChatFormProps &
       Partial<InsertAdditionalProps & UpdateAdditionalProps>
 
+  const setPostSubscriptionState = useSubscriptionState(
+    (state) => state.setPostSubscriptionState
+  )
+
+  const [newChatId, setNewChatId] = useState('')
```

Comment:
Can you explain a reason for doing it?

Reply:
datahub has some lag before the tx succeed and the post ready to be fetched, so before doing this, the user is redirected before the post is ready, so they see chat page with empty chat details

because of that, in this hack, it waits the post to be ready, so only after that, the user will be redirected